source/computer/wsl-setup.rst

```diff
@@ -1,18 +1,138 @@
+命令将整个子系统随时打包备份，并在之后利用 ``wsl --import`` 命令进行还原。与虚拟机相比，
+在启动速度、消耗资源以及性能利用上有较大优势。
+
+WSL 分为 WSL1 和 WSL2 两个发行版本。由于底层原理不同，在大多数情况下，建议使用 WSL2，
+因为它提供更快的性能和 100% 的系统调用兼容性。仅当你的项目需要经常从 Linux 上访问存储
+在 Windows 系统上的文件或者利用 Windows 软件对 Linux 上的文件进行处理时，通过 WSL1 可实现更快的性能。
+
+官方目前没有弃用 WSL1 的计划，并且支持将任何一个已经安装的 Linux 发行版更改为 WSL1 或者 WSL2。
+
+安装 WSL
+---------------
+
+WSL 的安装方式可以参考官方文档：
+
+- 中文指南：https://docs.microsoft.com/zh-cn/windows/wsl/install-win10
+- 英文指南：https://docs.microsoft.com/en-us/windows/wsl/install-win10
+
+官方文档对 WSL 安装配置中可能遇到的常见问题都给出了详细的解决方案，请认真阅读。
+
+WSL 可以安装不同的 Linux 发行版，但目前官方并未提供 Fedora 发行版的安装。
+推荐使用  WSL 安装 Ubuntu 20.04 LTS 或 Ubuntu 20.10。
+配置 Linux
+-----------
+
+按照官方文档安装 WSL 之后，就可以根据你的 Linux 发行版进行配置使用了。
 
-   尚未完成。
+如果你安装了 Ubuntu，可以参考 《\ :doc:`/computer/ubuntu-setup` 》进行配置。
 
-   WSL 的安装方式可以参考：
+图形界面显示
+--------------
 
-   - 中文指南：https://blog.walterlv.com/post/how-to-install-wsl2.html
-   - 英文指南：https://docs.microsoft.com/en-us/windows/wsl/install-win10
+由于 WSL 本身不支持图形界面，我们需要额外安装 X Server 并指定图形输出位置，使得带有图形界面的
+桌面程序可以被显示和运行。
+
+Windows 上多种可选的 X Server 软件，`VcXsrv <https://sourceforge.net/projects/vcxsrv/>`__ 
+作为开源免费软件，使用方式简单，本文以此软件为例进行介绍。
+
+1. 下载 `VcXsrv <https://sourceforge.net/projects/vcxsrv/>`__\ ，默认安装即可；
+
+2. 运行 XLaunch，除了在 ``Extra settings`` 界面勾选 ``Disable access control`` 之外，其它选项无需更改；
+
+3. 在 Linux 的 ``~/.bashrc`` 文件中添加如下命令后执行 ``source ~/.bashrc``：
+   
+::    
+
+    # Vcxsrv Display
+    # Windows 每次重启后 WSL2 namesever 的 IP 可能发生变化，如下设置保证能够连接到 XServer
+    export DISPLAY=$(awk '/nameserver / {print $2; exit}' /etc/resolv.conf 2>/dev/null):0
+    export LIBGL_ALWAYS_INDIRECT=1
+
+4、打开图形界面进行测试。
 
 .. note::
 
-   由于维护者目前没有可供安装 Windows 的空闲机器，因而本文短期内无法完成。
-   如果你有 Windows 使用经验，且愿意从头安装 WSL，请到
-   `Issue #65 <https://github.com/seismo-learn/seismology101/issues/65>`__
-   下发表评论。
+   安装并配置好 XServer 之后，切记先运行 XLaunch 再进入 Linux 环境打开图形界面。
+   
+WSL 重要命令
+---------------
+  
+以下命令均在 PowerShell 的管理员模式下运行，并且已安装 WSL2 版本的 Ubuntu20.04LTS，名称为 Ubuntu
+
+::    
+
+    # 查看 wsl 命令帮助
+    $ wsl -h
+    
+    # 列出所有已安装的 Linux 发行版的状态，并显示是 wsl1 还是 wsl2
+    $ wsl -l -v
+    
+    # 停止正在运行的 Linux 发行版，假设其名字为 Ubuntu
+    $ wsl -t Ubuntu
+    
+    # 将 Ubuntu 由 wsl2 更改为 wsl1
+    $ wsl --set-version Ubuntu 1
+    
+    # 将 Ubuntu 由 wsl1 改回 wsl2
+    $ wsl --set-version Ubuntu 2
+    
+    # 进入 Linux 环境
+    $ bash
+    
+    # 退出 Linux 环境
+    $ exit
+    
+    # 将名字为 Ubuntu 的 Linux 发行版导出备份
+    $ mkdir -p D:\WSLBAK  # 创建备份文件夹
+    $ wsl --export Ubuntu D:\WSLBAK\20210117bak.tar   # 备份 wsl 
+    
+    # 将之前备份的 Linux 发行版 Ubuntu 导入还原
+    # 默认的 Linux 发行版都安装在 C 盘，导入还原的时候可以选择其它位置避免 C 盘空间不够
+    $ mkdir -p D:\WSLDIR\Ubuntu20.04  # 创建还原的位置
+    # 还原 wsl 到 D 盘，并命名为 Ubuntu20.04
+    $ wsl --import Ubuntu20.04 D:\WSLDIR\Ubuntu20.04 D:\WSLBAK\20210117bak.tar  
+    
+    # 设置默认发行版
+    # 因为现在有不止一个 Linux 发行版存在，所以需要选择键入 bash 的时候激活哪个
+    $ wsl -s Ubuntu20.04   
+    
+    # 删除某个发行版
+    $ wsl --unregister Ubuntu  # 删除名为 Ubuntu 的发行版
+
```

Comment:
Do we want to split those commands in different code windows instead of a single one?

Reply:
I think it's clearly enough.

Reply:
Yes, the content is clearly enough. 

I mean could we add 2 or 3 classifications so that new babies will have a better visual feeling.

For example, the first five commands are related to wsl, while the last four are related to operate
Linux distribution. Meanwhile, "Linux 环境" has two related commands.

Reply:
To me, the first 5 and the last 4 commands belong to the same group, showing the usage of `wsl` command. The `bash` and `exit` commands belong to another group.

Maybe two groups are enough?

Reply:
> To me, the first 5 and the last 4 commands belong to the same group, showing the usage of `wsl` command. The `bash` and `exit` commands belong to another group.
> 
> Maybe two groups are enough?

Two groups are okay. The order may be changed a little.